spring-context/src/main/java/org/springframework/validation/Validator.java

```diff
@@ -92,4 +98,39 @@ public interface Validator {
 	 */
 	void validate(Object target, Errors errors);
 
+	/**
+	 * Returns the {@link Function} that takes the {@link BiConsumer} containing validation logic for the specific type
+	 * <code>&lt;T&gt;</code> and returns the {@link Validator} instance.<br>
+	 * This validator implements the <i>typical</i> {@link #supports(Class)} method
+	 * for the given <code>&lt;T&gt;</code>.<br>
+	 *
+	 * By using this {@link #of(Class)} method, a {@link Validator}  can be implemented as follows:
+	 *
+	 * <pre class="code">Validator passwordEqualsValidator = Validator.of(PasswordResetForm.class)
+	 *     .apply((form, errors) -> {
+	 *       if (!Objects.equals(form.getPassword(), form.getConfirmPassword())) {
+	 *         errors.rejectValue("confirmPassword",
+	 *             "PasswordEqualsValidator.passwordResetForm.password",
+	 *             "password and confirm password must be same.");
+	 *       }
+	 *     });</pre>
+	 * @param targetClass  the class of the object that is to be validated
+	 * @param <T> the type of the object that is to be validated
+	 * @return the {@link Function} that takes the {@link BiConsumer} containing the validation logic and
+	 * returns the {@link Validator} instance
+	 */
+	static <T> Function<BiConsumer<T, Errors>, Validator> of(Class<T> targetClass) {
+		Assert.notNull(targetClass, "'targetClass' must not be null.");
+		return validator -> new Validator() {
```

Comment:
Variable name `validator` is a bit confused here since it's not type of `Validator` but `BiConsumer`.

Reply:
Indeed, the naming is a bit tricky.

That `BiConsumer` effectively implements the `validate(Object, Errors)` method from `Validator`. So, it's more like "validation logic" or "validation delegate". 

Anyone have any good suggestions?

Reply:
renamed to `delegate`